### PR TITLE
Table comparison: exclude structures

### PR DESCRIPTION
**Pull request by christianguenter2**
#1334

### CODE REVIEW

**Comment by larshp** [comment]
yeah

I'm not sure how to handle this, the structure could also be in use in a database table via .INCLUDE

Perhaps look through 5 levels of where-used, to see if it is part of a database table. Guess it is only relevant to look for structure -> structure^n -> database table relations

**Comment by christianguenter2** [comment]
good point. Maybe something like this?

File: src/objects/zcl_abapgit_object_tabl_valid.clas.abap [hunk 1]
```diff
@@ -8,20 +8,132 @@ CLASS zcl_abapgit_object_tabl_valid DEFINITION PUBLIC FINAL.
...
+          batchjob_error           = 6
+          wrong_type               = 7
+          object_not_exist         = 8
+          OTHERS                   = 9.
+
+      IF sy-subrc = 1 OR lines( lt_founds ) = 0.
+        EXIT.
+      ENDIF.
+
+      IF sy-subrc > 1.
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_tabl_valid.clas.abap]: use ELSEIF

```abap
      ENDIF.

      IF sy-subrc > 1.
```

File: src/objects/zcl_abapgit_object_tabl_valid.clas.abap [hunk 2]
```diff
@@ -8,20 +8,132 @@ CLASS zcl_abapgit_object_tabl_valid DEFINITION PUBLIC FINAL.
...
+
+    IF ls_dd02v-tabclass <> 'TRANSP'.
+      " We only want to compare transparent tables, or structures
+      " used in transparent tables
+
+      IF is_structure_used_in_db_table( ls_dd02v-tabname ) = abap_false.
+        RETURN.
+      ENDIF.
+
+    ENDIF.
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_tabl_valid.clas.abap]: IF in IF, use AND

**Pull request merged**